src/rt/ds/hashmap.h

```diff
@@ -308,7 +308,7 @@ namespace verona::rt
 
       assert(key_of(entry) != 0);
       const auto key = unmark_key(key_of(entry));
-      const auto hash = bits::hash((void*)key);
+      const auto hash = bits::hash(((const Object*)key)->id());
```

Comment:
I assume key is guaranteed to be an Object, then a c++ static cast would be more appropriate.

Is there any compile time expression (template parameter, type trait, etc) checking to make sure it is?

Reply:
Yes, there are: https://github.com/microsoft/verona/blob/master/src/rt/ds/hashmap.h#L54